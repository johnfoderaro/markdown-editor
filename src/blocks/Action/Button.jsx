import styled from 'styled-components';

import Button from '../../elements/Button';

const colors = (type) => {
  switch (type) {
    case 'new':
      return 'rgb(25, 105, 179)';
    case 'save':
      return 'rgb(25, 179, 63)';
    case 'delete':
      return 'rgb(179, 56, 25)';
    default: // 'inverse':
      return 'rgb(255, 255, 255)';
  }
};

const StyledButton = styled(Button)`
  color: ${({ type }) => (type !== 'inverse' ? 'inherit' : 'rgb(255, 255, 255)')};
  margin-right: ${({ type }) => {
    switch (type) {
      case 'new':
        return '.5rem';
      case 'save':
        return '.5rem';
      default:
        return '0';
    }
  }};
  margin-left: ${({ type }) => {
    switch (type) {
      case 'delete':
        return '.5rem';
      default:
        return '0';
    }
  }};
  border: 2px solid ${({ type }) => colors(type)};
    &:hover {
      border: none;
      background: ${({ type }) => colors(type)};
      color: ${({ type }) => (type === 'inverse' ? 'rgb(0, 0, 0)' : 'rgb(255, 255, 255)')};
  }
`;

export default StyledButton;
