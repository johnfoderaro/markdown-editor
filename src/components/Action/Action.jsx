import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';

// import Text from '../../blocks/Text';
import ButtonGroup from '../../blocks/ButtonGroup';

const Editor = ({ file }) => (
  <ButtonGroup />
);

Editor.propTypes = {
  file: PropTypes.string,
};

Editor.defaultProps = {
  file: PropTypes.string,
};
export default hot(module)(Editor);
