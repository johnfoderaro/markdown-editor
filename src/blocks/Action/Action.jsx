import styled from 'styled-components';

import Button from './Button';

const Action = styled.div`
  margin-top: auto;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`;

Action.Button = Button;

export default Action;
