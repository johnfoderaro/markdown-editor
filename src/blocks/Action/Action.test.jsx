import React from 'react';
import { shallow } from 'enzyme';

import Action from './Action';

let tree;

beforeAll(() => {
  tree = shallow(
    <Action>
      <Action.Button type="new">
        New
      </Action.Button>
      <Action.Button type="save">
        Save
      </Action.Button>
      <Action.Button type="delete">
        Delete
      </Action.Button>
    </Action>,
  );
});

describe('<Action />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
    expect(tree.find(Action.Button).at(0).props().type).toEqual('new');
    expect(tree.find(Action.Button).at(1).props().type).toEqual('save');
    expect(tree.find(Action.Button).at(2).props().type).toEqual('delete');
  });
  it('should contain a <Action.Button /> block', () => {
    tree.find(Action.Button).at(0).simulate('hover');
  });
});
