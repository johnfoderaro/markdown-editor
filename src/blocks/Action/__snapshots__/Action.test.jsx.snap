// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef>
    <ForwardRef
      type="new"
    >
      New
    </ForwardRef>
    <ForwardRef
      type="save"
    >
      Save
    </ForwardRef>
    <ForwardRef
      type="delete"
    >
      Delete
    </ForwardRef>
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ForwardRef
          type="new"
        >
          New
        </ForwardRef>,
        <ForwardRef
          type="save"
        >
          Save
        </ForwardRef>,
        <ForwardRef
          type="delete"
        >
          Delete
        </ForwardRef>,
      ],
      "forwardedClass": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Button": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  .editor .action-button.save {
    margin-right: ",
              [Function],
              ";
    margin-left: ",
              [Function],
              ";
    border: 2px solid ",
              [Function],
              ";
    &:hover {
      border: none;
      background: ",
              [Function],
              ";
      color: rgb(255, 255, 255);
    }
  }
",
            ],
          },
          "displayName": "Styled(Button)",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  margin-top: auto;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "New",
          "type": "new",
        },
        "ref": null,
        "rendered": "New",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  .editor .action-button.save {
    margin-right: ",
              [Function],
              ";
    margin-left: ",
              [Function],
              ";
    border: 2px solid ",
              [Function],
              ";
    &:hover {
      border: none;
      background: ",
              [Function],
              ";
      color: rgb(255, 255, 255);
    }
  }
",
            ],
          },
          "displayName": "Styled(Button)",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Save",
          "type": "save",
        },
        "ref": null,
        "rendered": "Save",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  .editor .action-button.save {
    margin-right: ",
              [Function],
              ";
    margin-left: ",
              [Function],
              ";
    border: 2px solid ",
              [Function],
              ";
    &:hover {
      border: none;
      background: ",
              [Function],
              ";
      color: rgb(255, 255, 255);
    }
  }
",
            ],
          },
          "displayName": "Styled(Button)",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "type": "delete",
        },
        "ref": null,
        "rendered": "Delete",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  .editor .action-button.save {
    margin-right: ",
              [Function],
              ";
    margin-left: ",
              [Function],
              ";
    border: 2px solid ",
              [Function],
              ";
    &:hover {
      border: none;
      background: ",
              [Function],
              ";
      color: rgb(255, 255, 255);
    }
  }
",
            ],
          },
          "displayName": "Styled(Button)",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ForwardRef
            type="new"
          >
            New
          </ForwardRef>,
          <ForwardRef
            type="save"
          >
            Save
          </ForwardRef>,
          <ForwardRef
            type="delete"
          >
            Delete
          </ForwardRef>,
        ],
        "forwardedClass": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Button": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  .editor .action-button.save {
    margin-right: ",
                [Function],
                ";
    margin-left: ",
                [Function],
                ";
    border: 2px solid ",
                [Function],
                ";
    &:hover {
      border: none;
      background: ",
                [Function],
                ";
      color: rgb(255, 255, 255);
    }
  }
",
              ],
            },
            "displayName": "Styled(Button)",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  margin-top: auto;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "New",
            "type": "new",
          },
          "ref": null,
          "rendered": "New",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  .editor .action-button.save {
    margin-right: ",
                [Function],
                ";
    margin-left: ",
                [Function],
                ";
    border: 2px solid ",
                [Function],
                ";
    &:hover {
      border: none;
      background: ",
                [Function],
                ";
      color: rgb(255, 255, 255);
    }
  }
",
              ],
            },
            "displayName": "Styled(Button)",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Save",
            "type": "save",
          },
          "ref": null,
          "rendered": "Save",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  .editor .action-button.save {
    margin-right: ",
                [Function],
                ";
    margin-left: ",
                [Function],
                ";
    border: 2px solid ",
                [Function],
                ";
    &:hover {
      border: none;
      background: ",
                [Function],
                ";
      color: rgb(255, 255, 255);
    }
  }
",
              ],
            },
            "displayName": "Styled(Button)",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "type": "delete",
          },
          "ref": null,
          "rendered": "Delete",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  .editor .action-button.save {
    margin-right: ",
                [Function],
                ";
    margin-left: ",
                [Function],
                ";
    border: 2px solid ",
                [Function],
                ";
    &:hover {
      border: none;
      background: ",
                [Function],
                ";
      color: rgb(255, 255, 255);
    }
  }
",
              ],
            },
            "displayName": "Styled(Button)",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
